modules/contact/src/actions/ContactAction.C

```diff
@@ -51,6 +51,10 @@ validParams<ContactAction>()
       "penalty",
       1e8,
       "The penalty to apply.  This can vary depending on the stiffness of your materials");
+  params.addParam<Real>("tangential_penalty",
+                        1e8,
+                        "The tangential penalty to apply for frictional contact.  This can vary "
+                        "depending on the stiffness of your materials");
```

Comment:
Getting these parameters right is a can of worms. A few issues here:
1. We should probably disallow the use of `tangential_penalty` for frictionless and glued contact because it doesn't do anything in those case, so it's misleading to provide it as an option.
2. Default values for either `tangential_penalty` or `penalty` are problematic because they are problem-dependent. Maybe we should make `penalty` required, and have an optional multiplier on that for the tangential component that defaults to 1. Auto scaling makes this even messier.
3. The descriptions for 'penalty' and 'tangential_penalty' aren't very helpful. We can't explain all the intricacies of how to use them in different scenarios in one line, but we need to make sure that is covered in the .md document, and we should do better here too.

Reply:
>     2\. Maybe we should make `penalty` required, and have an optional multiplier on that for the tangential component that defaults to 1.

I think we should be moving away from use of `penalty` in input files, not making it required (obviously with the exception of when we are actually using a `ContactFormulation::PENALTY`). If the user is using automatic scaling, as of this PR we error out if the user sets a non-unity penalty for kinematic formulations. Requiring them to set a parameter that we require to be 1 in a lot of cases seems silly. Moreover, mortar contact and RANFS contact do not use penalty parameters at all. I would suggest that if we were to change anything for the `penalty` parameter it would be to make its default value 1. I think `tangential_penalty` should be its own parameter because in the future I would love to see an input block for a `tangential_penalty` formulation that has no `penalty` parameter at all since the user shouldn't have to set it if they are using automatic scaling.

Reply:
>     3\. The descriptions for 'penalty' and 'tangential_penalty' aren't very helpful. We can't explain all the intricacies of how to use them in different scenarios in one line, but we need to make sure that is covered in the .md document, and we should do better here too.

`tangential_penalty` isn't that complicated is it? It always corresponds to a true penalty concept right? Or is there something weird for Augmented Lagrange? I vaguely remember you saying something related to that. I don't know anything about `tangential_penalty` other than that it shouldn't be equal to the `penalty` for a `TANGENTIAL_PENALTY` formulation

Reply:
In the past, have you typically used the same order of magnitude `penalty` for both `KINEMATIC` and `PENALTY`?

Reply:
Yes, in the past they have been of a similar order of magnitude.  I'm just saying here that since these parameters are used different ways for the different formulations, it's more than you can explain in a short parameter description.  Also keep in mind that these behave differently when you use the normalize_penalty option, which also needs to be explained.  That sentence about it varying based on the stiffness of the materials isn't terribly descriptive in my opinion.

Reply:
> That sentence about it varying based on the stiffness of the materials isn't terribly descriptive in my opinion.

But I feel like if you're going to give a one sentence description, that sentence is not a bad one. The only way I would improve it given a concision constraint is to say this would be roughly on the order of magnitude of the stiffness.